src/Microsoft.HttpRepl.IntegrationTests/Preferences/UserFolderPreferencesTests.cs

```diff
@@ -82,37 +81,159 @@ public void WritePreferences_ChangeNonDefaultToDefault_RemovesDefaultValue()
         {
             string originalValue = "BoldMagenta";
             string defaultValue = "Red";
-            IReadOnlyDictionary<string, string> defaultPreferences = new Dictionary<string, string> { { WellKnownPreference.ProtocolColor, defaultValue } };
+            IDictionary<string, string> defaultPreferences = new Dictionary<string, string> { { WellKnownPreference.ProtocolColor, defaultValue } };
 
-            SetupPreferences(out UserFolderPreferences preferences, out MockedFileSystem fileSystem);
+            MockedFileSystem fileSystem = new MockedFileSystem();
+            IUserProfileDirectoryProvider userProfileDirectoryProvider = new UserProfileDirectoryProvider();
+            UserFolderPreferences preferences = new UserFolderPreferences(fileSystem, userProfileDirectoryProvider, defaultPreferences);           
 
             // Create a file with a non-default value, read it from the file system and
             // validate that it was read correctly
             fileSystem.AddFile(preferences.PreferencesFilePath, $"{WellKnownPreference.ProtocolColor}={originalValue}");
-            Dictionary<string, string> preferencesFromFile = preferences.ReadPreferences(defaultPreferences);
 
-            Assert.Equal(originalValue, preferencesFromFile[WellKnownPreference.ProtocolColor]);
+            Assert.Equal(originalValue, preferences.CurrentPreferences[WellKnownPreference.ProtocolColor]);
 
             // Now change it to the default value, write it back to the file system and
             // validate that it was removed from the file
-            preferencesFromFile[WellKnownPreference.ProtocolColor] = defaultPreferences[WellKnownPreference.ProtocolColor];
-            bool succeeded = preferences.WritePreferences(preferencesFromFile, defaultPreferences);
+            bool succeeded = preferences.SetValue(WellKnownPreference.ProtocolColor, defaultPreferences[WellKnownPreference.ProtocolColor]);
 
             Assert.True(succeeded);
             Assert.Equal(string.Empty, fileSystem.ReadFile(preferences.PreferencesFilePath));
         }
 
+        [Fact]
+        public void GetColorValue_DoesNotExist_ReturnsDefault()
+        {
+            ConfirmColorValue(expected: AllowedColors.Magenta,
```

Comment:
If all these tests are really identical except for inputs, use `[Theory]` instead of `[Fact]`.  This also helps keep the test body in AAA form without having to jump elsewhere to read the code.

Reply:
Done in b318acc, though I'll admit its annoying to not be able to specify a test name for each set of data.